File: app/src/main/java/com/manoj/dlt/interfaces/IDeepLinkHistory.java
package com.manoj.dlt.interfaces;

import com.manoj.dlt.models.DeepLinkInfo;

import java.util.List;

public interface IDeepLinkHistory
{
    public void addLinkToHistory(DeepLinkInfo deepLinkInfo);

    public void removeLinkFromHistory(String deepLink);

    public void clearAllHistory();
}

File: app/src/main/kotlin/com/manoj/dlt/interfaces/IDeepLinkHistory.kt
package com.manoj.dlt.interfaces

import com.manoj.dlt.models.DeepLinkInfo

interface IDeepLinkHistory {
    fun addLinkToHistory(deepLinkInfo: DeepLinkInfo)

    fun removeLinkFromHistory(deepLinkId: String)

    fun clearAllHistory()

    val linkHistoryFromFileSystem: List<DeepLinkInfo>
}
